geomstats/geometry/complex_poincare_disk.py

```diff
@@ -0,0 +1,360 @@
+"""The complex Poincaré disk manifold.
+
+The Poincaré complex disk is a representation of the Hyperbolic space of dimension 2.
+
+Lead author: Yann Cabanes.
+
+References
+----------
+.. [Cabanes2022] Yann Cabanes. Multidimensional complex stationary
+    centered Gaussian autoregressive time series machine learning
+    in Poincaré and Siegel disks: application for audio and radar
+    clutter classification, PhD thesis, 2022
+.. [JV2016] B. Jeuris and R. Vandebril. The Kahler mean of Block-Toeplitz
+    matrices with Toeplitz structured blocks, 2016.
+    https://epubs.siam.org/doi/pdf/10.1137/15M102112X
+.. [Yang2013] Marc Arnaudon, Frédéric Barbaresco and Le Yang.
+    Riemannian medians and means with applications to radar signal processing,
+    IEEE Journal of Selected Topics in Signal Processing, vol. 7, no. 4, pp. 595-604,
+    Aug. 2013, doi: 10.1109/JSTSP.2013.2261798.
+    https://ieeexplore.ieee.org/document/6514112
+"""
+
+import geomstats.backend as gs
+from geomstats.geometry.base import ComplexOpenSet
+from geomstats.geometry.complex_riemannian_metric import ComplexRiemannianMetric
+from geomstats.geometry.hermitian import Hermitian
+
+
+class ComplexPoincareDisk(ComplexOpenSet):
+    """Class for the complex Poincaré disk.
+
+    The Poincaré disk is a representation of the Hyperbolic
+    space of dimension 2. Its complex dimension is 1.
+
+    Parameters
+    ----------
+    scale : float
+        Scale of the complex Poincaré metric.
+        Optional, default: 1.
+    """
+
+    def __init__(self, scale=1.0, **kwargs):
+        super().__init__(
+            dim=1,
+            embedding_space=Hermitian(dim=1),
+            metric=ComplexPoincareDiskMetric(scale=scale),
+            **kwargs
+        )
+        self.scale = scale
+
+    @staticmethod
+    def belongs(point, atol=gs.atol):
+        """Check if a point belongs to the complex unit disk.
+
+        Check if a point belongs to the Poincaré complex disk,
+        i.e. Check if its norm is lower than one.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., 1]
+            Point to be checked.
+        atol : float
+            Tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean denoting if point belongs to the
+            complex Poincaré disk.
+        """
+        is_scalar = gs.ndim(point) == 1
+        is_scalar = is_scalar or (gs.ndim(point) == 2 and point.shape[1] == 1)
```

Comment:
I don't think the second logical statement should work: we are not expecting point to be 2d (in batch mode it is 1d).